src/rhsmlib/services/unregister.py

```diff
@@ -0,0 +1,74 @@
+from __future__ import print_function, division, absolute_import
+
+#
+# Copyright (c) 2017 Red Hat, Inc.
+#
+# This software is licensed to you under the GNU General Public License,
+# version 2 (GPLv2). There is NO WARRANTY for this software, express or
+# implied, including the implied warranties of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+# along with this software; if not, see
+# http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+#
+# Red Hat trademarks are not licensed under GPLv2. No permission is
+# granted to use or replicate Red Hat trademarks that are incorporated
+# in this software or its documentation.
+#
+
+"""
+This module includes class used for unregistering system from candlepin
+server.
+"""
+
+import logging
+
+from subscription_manager import injection as inj
+from subscription_manager import managerlib
+from rhsm import connection
+
+
+log = logging.getLogger(__name__)
+
+
+class UnRegister(object):
+    """
+    Class providing functionality of unregistering the system from
+    Candlepin server.
+    """
+
+    def __init__(self, uep, consumer_uuid):
```

Comment:
I wouldn't pass in the consumer_uuid.  I can't envision a scenario where subscription-manager is going to try to unregister anyone but itself.  Today the CLI only allows a system to unregister itself.  I think it'd make more sense to grab an Identity object via injection in `unregister()` and then grab the consumer_uuid there.